### ISSUE TITLE
Detect on install which system/os and app the addon gets installed to

### ISSUE THREAD

**Issue by tooomm** [mkiol/GNotifier#135]
Display/hide settings in options accordingly and don't mess around with the user too much. :D
On Firefox (win+mac) for example 80% of settings are not relevant...

![settings_limitations](https://cloud.githubusercontent.com/assets/9874850/18617411/efef3930-7dce-11e6-973a-fd68fb8d0365.png)

**Comment by mkiol** [comment]
I agree. It is a mess. Unfortunately, there is no super easy way how to make it more clear. Let's keep it open and maybe I will figure something out in the near future...

**Comment by tooomm** [comment]
Detecting the needed information should be possible regarding this:
https://developer.mozilla.org/en-US/Add-ons/SDK/High-Level_APIs/system

```
console.log(system.env.PATH);
// operating system
```
and
```
console.log("id = " + system.id);
// host application name
```

<br>
I'm not a developer or such... you need to double check and validate for sure. But it sounds reasonable, there are many more variables listed on that page which might be helpful in that case.

The addon then would require to just display options which are relevant (or hide others). That might be a different problem. I don't know.

What do you think?

Anyway, I appreciate that you are aware of this and that you keep it in mind.^^

**Comment by mkiol** [comment]
Thank you for the hints. The problem is that for the preferences I decided to use built-in FF mechanism. I just defined various options using [package.json](https://github.com/mkiol/GNotifier/blob/master/source/package.json) file and FF is responsible for the presentation. Unfortunately with this approach, there is no place where I can insert some code that will hide/show options etc. :-(

I see two potential solutions:
- create my own preferences window where I can code whatever I want, or
- create many XPIs with different package.json for: Linux, Window, Firefox, Thunderbird, Seamonkey

**Comment by tooomm** [comment]
Oh, ok. That makes sense...

The second options sounds like a mess though, and much work to maintain and supply properly.

**Comment by rcmero** [comment]
About this issue, maybe you could ask for help from the developer of the Classic Toolbar Buttons extension? That extension also uses the built-in FF mechanism, and the options are diferent depending on whether the addon is installed on Firefox, Thunderbird or SeaMonkey.

**Comment by tooomm** [comment]
Now with this add-on beeing Thunderbird only (#162) you can remove a lot of the `(Thunderbird only)` hints for future versions!
